core/test_network/src/lib.rs

```diff
@@ -662,11 +665,12 @@ impl PeerBuilder {
 
 type PeerWithRuntimeAndClient = (Runtime, Peer, Client);
 
-fn local_unique_port() -> Result<String> {
-    Ok(format!(
-        "127.0.0.1:{}",
-        unique_port::get_unique_free_port().map_err(Error::msg)?
-    ))
+fn local_unique_port() -> Result<SocketAddr> {
+    Ok(SocketAddrV4 {
+        ip: Ipv4Addr([127, 0, 0, 1]),
```

Comment:
If this is why you made the bytes public, I'd recommend using a constructor through a macro.